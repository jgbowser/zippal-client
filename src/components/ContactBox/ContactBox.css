.contact_card_box {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.contact_card {
  width: 15vw;
  background-color: burlywood;
  min-width: 200px;
}
.ContactDetails {
  width: 70%;
  margin: 0 auto;
  padding-bottom: 20px;
  margin-bottom: 70px;
  background-image: linear-gradient(360deg, white, darkkhaki);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.ContactDetails h4 {
  font-size: 28px;
}

.ContactDetails span {
  margin-left: 10px;
}

.ContactDetails a {
  text-decoration: none;
  color: black;
  font-size: 24px;
  margin: 10px;
  padding: 10px;
  border-radius: 10px;
}

.ContactDetails a:hover {
  color: blue;
  background-color: darkkhaki;
}

.contact_img {
  width: 150px;
  margin: 0 auto;
}

.contact_img img {
  width: 100%;
  border-radius: 50%;
}

.personal_bio {
  padding: 0 40px;
}

.contact_card_box button {
  margin: 0;
  padding: 5px;
  width: 22%;
  background-color: darkkhaki;
  box-shadow: 0px transparent;
  border: 1px solid #6c683e;
  box-shadow: 0px 0px 0px transparent;
  text-shadow: 0px 0px 0px transparent;
}

.contact_card_box button:hover {
  background: #6c683e;
  box-shadow: 0px 0px 0px transparent;
  border: 0px solid transparent;
  text-shadow: 0px 0px 0px transparent;
  color: #d6d8e3;
  cursor: pointer;
}

.contact_card_box button:active {
  outline: none;
  border: none;
}

.contact_card_box button:focus {
  outline: 0;
  background: #beb442;
  color: #223187;
}
