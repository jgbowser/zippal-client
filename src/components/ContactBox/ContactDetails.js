import React from 'react'

export function JohnDetails () {
  return (
    <div>

    </div>
  )
}

export function RyanDetails () {
  return (
    <div>
      <h4>Ryan Whtimore</h4>
      Github: <a href="https://github.com/warptrail"> warptrail</a>
    </div>
  )
}

