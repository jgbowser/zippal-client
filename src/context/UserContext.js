import React, {Component} from 'react'
import AuthApiService from '../services/auth-api-service'
import UserService from '../services/user-api-service'
import TokenService from '../services/token-service'
import IdleService from '../services/idle-service'

const UserContext = React.createContext({
  user:{},
  profileInfo:{},
  error:null,
  setError:() => { },
  clearError: () => { },
  setUser: () => { },
  getProfile: () => { },
  updateProfile: () => { },
  processLogin: () => { },
  processLogout: () => { },
})

export default UserContext

export class UserProvider extends Component {
  constructor(props){
    super(props)
    const state = {user: {}, error:null, profileInfo: {}}

    const jwtPayload = TokenService.parseAuthToken()
    

    if(jwtPayload){
      const profileData = this.getProfile()

      state.user = {
        id: jwtPayload.id,
        display_name: jwtPayload.display_name,
        username: jwtPayload.sub,
      }

      if(profileData)
        state.profileInfo = {
          bio: profileData.bio,
          display_name: profileData.display_name,
          fa_icon: profileData.fa_icon,
          location: profileData.location
      }
    }
      

    this.state = state;
    IdleService.setIdleCallback(this.logoutBecauseIdle)
  }

  componentDidMount(){
    if(TokenService.hasAuthToken()){
      IdleService.registerIdleTimerResets()
      TokenService.queueCallbackBeforeExpiry(() => {
        this.fetchRefreshToken()
      })
    }
  }

  componentWillUnmount() {
    IdleService.unRegisterIdleResets()
    TokenService.clearCallBackBeforeExpiry()
  }

  setError = error => {
    console.error(error)
    this.setState({error})
  }

  clearError = () => {
    this.setState({error: null})
  }
  
  setUser = user => {
    this.setState({user})
  }

  getProfile = async () => {
    const profileInfo = await UserService.getUserProfile()
    this.setState({profileInfo})
  }

  updateProfile = (profileInfo) => {
    this.setState({profileInfo})
  }

  processLogin = authToken => {
    TokenService.saveAuthToken(authToken)

    const jwtPayload = TokenService.parseAuthToken()
    this.setUser({
      id: jwtPayload.id,
      display_name: jwtPayload.display_name,
      username: jwtPayload.sub,
    })
    this.getProfile()
    IdleService.registerIdleTimerResets()
    TokenService.queueCallbackBeforeExpiry(() => {
      this.fetchRefreshToken()
    })
  }

  processLogout = () => {
    TokenService.clearAuthToken()
    TokenService.clearCallBackBeforeExpiry()
    IdleService.unRegisterIdleResets()
    this.setUser({})
    this.setState({profileInfo: {}})
  }

  logoutBecauseIdle = () => {
    TokenService.clearAuthToken()
    TokenService.clearCallBackBeforeExpiry()
    IdleService.unRegisterIdleResets()
    this.setUser({idle:true})
  }

  fetchRefreshToken = () => {
    AuthApiService.refreshToken()
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        TokenService.queueCallbackBeforeExpiry(() => {
          this.fetchRefreshToken()
        })
      })
      .catch(err => {
        this.setError(err)
      })
  }

  render() {
    const value = {
      user: this.state.user,
      profileInfo: this.state.profileInfo,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setUser: this.setUser,
      getProfile: this.getProfile,
      updateProfile: this.updateProfile,
      processLogin: this.processLogin,
      processLogout: this.processLogout
    }
    return (
      <UserContext.Provider value={value}>
        {this.props.children}
      </UserContext.Provider>
    )
  }
}