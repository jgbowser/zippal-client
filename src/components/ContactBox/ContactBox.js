/* eslint-disable default-case */
import React, {Component} from 'react'
import {RyanDetails} from './ContactDetails'
import './ContactBox.css'

 export default class ContactBox extends Component {
  state = {
    johnToggled: false,
    philipToggled: false,
    MathewToggled: false,
    ryanToggled: false
  }

  toggleJohn = () => {
    this.setState({
      johnToggled: !this.state.johnToggled,
      philipToggled: false,
      MathewToggled: false,
      ryanToggled: false
    })
  }
  togglePhilip = () => {
    this.setState({
      johnToggled: false,
      philipToggled: !this.state.philipToggled,
      MathewToggled: false,
      ryanToggled: false
    })
  }
  toggleMathew = () => {
    this.setState({
      johnToggled: false,
      philipToggled: false,
      MathewToggled: !this.state.MathewToggled,
      ryanToggled: false
    })
  }
  toggleRyan = () => {
    this.setState({
      johnToggled: false,
      philipToggled: false,
      MathewToggled: false,
      ryanToggled: !this.state.ryanToggled
    })
  }



  renderDetails = () =>{
    let nameToggled = ''
    let nameDetails = ''
    let obj = this.state

    for (const key of Object.keys(obj)){
      if(obj[key] === true){
        nameToggled = key
      }
    }

    switch(nameToggled) {
      case 'johnToggled':
        nameDetails = (
        <div>
          Hello John
        </div>
        )
        break
      case 'philipToggled':
        nameDetails = (
        <div>
          Hello Philip
        </div>
        )
        break
      case 'MathewToggled':
        nameDetails = (
        <div>
          Hello Mathew
        </div>
        )
        break
      case 'ryanToggled':
        nameDetails = <RyanDetails />
        break
      default:
        break
    }
    return nameDetails;
  }
  
  render(){
    this.renderDetails()
     return (
       <>
      <div className="contact_card_box">
      <button className="contact_card" onClick={this.toggleJohn}>
        <h3>John</h3>
      </button>
      <button className="contact_card" onClick={this.togglePhilip}>
        <h3>Philip</h3>
      </button>
      <button className="contact_card" onClick={this.toggleMathew}>
        <h3>Mathew</h3>
      </button>
      <button className="contact_card" onClick={this.toggleRyan}>
        <h3>Ryan</h3>
      </button>
    </div>
    <div className="contact_details">
    {this.renderDetails()}
    </div>
    </>
     )
   }
 }