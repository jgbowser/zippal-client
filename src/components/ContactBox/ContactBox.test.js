import React from 'react'
import ReactDOM from 'react-dom'
import testHelpers from '../../services/test-helpers'
import ContactBox from './ContactBox'
import UserContext from '../../context/UserContext'
import renderer from 'react-test-renderer'

describe.only('ContactBox Component', () => {
  

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <UserContext.Provider value={{user: {id: 1}}}>
        <ContactBox />
      </UserContext.Provider>, div
    )
    ReactDOM.unmountComponentAtNode(div)
  });
  it(`takes a snapshot of component`,() => {
    const tree = renderer
    .create(
      <UserContext.Provider value={{user:{id:1}}}>
        <ContactBox />
      </UserContext.Provider>
    )
    expect(tree).toMatchSnapshot();
  })
})