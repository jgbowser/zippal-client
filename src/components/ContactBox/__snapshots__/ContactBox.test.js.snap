// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactBox Component takes a snapshot of component 1`] = `
Array [
  <div
    className="contact_card_box"
  >
    <button
      className="contact_card"
      onClick={[Function]}
    >
      <h3>
        John
      </h3>
    </button>
    <button
      className="contact_card"
      onClick={[Function]}
    >
      <h3>
        Phillip
      </h3>
    </button>
    <button
      className="contact_card"
      onClick={[Function]}
    >
      <h3>
        Mathew
      </h3>
    </button>
    <button
      className="contact_card"
      onClick={[Function]}
    >
      <h3>
        Ryan
      </h3>
    </button>
  </div>,
  "",
]
`;
